app-modules/report/tests/Tenant/Filament/Widgets/MostEngagedStudentsTableTest.php

```diff
@@ -147,6 +150,48 @@
         'filters' => [],
     ])
         ->assertCanSeeTableRecords(collect([
-            $student1, $student2,
+            $student1,
+            $student2,
         ]));
 });
+
+it('has table an export action', function () {
+    livewire(MostEngagedStudentsTable::class, [
+        'cacheTag' => 'report-student-messages-overview',
+        'filters' => [],
+    ])->assertTableActionExists(ExportAction::class);
+});
+
+it('can start an export, sending a notification', function () {
+    asSuperAdmin();
+    Storage::fake('s3');
+    $count = random_int(1, 5);
+    $student1 = Student::factory()->create();
+    $student2 = Student::factory()->create();
+    $student3 = Student::factory()->create();
+
+    Engagement::factory()->count($count)->state([
+        'recipient_id' => $student1->sisid,
+        'recipient_type' => (new Student())->getMorphClass(),
+        'channel' => NotificationChannel::Email,
+    ])->create();
+
+    Engagement::factory()->count($count)->state([
+        'recipient_id' => $student2->sisid,
+        'recipient_type' => (new Student())->getMorphClass(),
+        'channel' => NotificationChannel::Email,
+    ])->create();
+
+    Engagement::factory()->count($count)->state([
+        'recipient_id' => $student3->sisid,
+        'recipient_type' => (new Student())->getMorphClass(),
+        'channel' => NotificationChannel::Email,
+    ])->create();
+
+    livewire(MostEngagedStudentsTable::class, [
+        'cacheTag' => 'report-student-engagement',
+        'filters' => [],
+    ])
+        ->callTableAction(ExportAction::class)
+        ->assertNotified();
```

Comment:
Is there any sort of check we can do here to ensure there were no errors? I believe there is a `->assertHasNoTableActionErrors()` or something like that?